builds.sh

```diff
@@ -130,11 +140,18 @@ for sourceId; do
 				end
 			')"
 
+			if [ -n "imageArchResolved["$lookup"]:+x" ]; then
```

Comment:
What is this line doing? From the following lines, it looks like you want to check if the value for the key `$lookup` of the associative array `imageArchResolved` has no value since when true you calculate the value and cache it and when false you just use the cached value. If this is the case, I would expect such test to look like this:

```sh
if [ -z "${imageArchResolved[$lookup]:+x}" ]; then
```

Note switching from `-n` to `-z` and and wrapping the evaluation of the associative array key and `:+` in `${}`.

As it stands now, this conditional will either always be true or error, at least if my internal bash parser is correct. It will be true if `$lookup` is zero or one "word" and will error if `$lookup` is two or more words. Since you are not dereferencing the `imageArchResolved` variable, i.e., as `${imageArchResolved[KEY]}`, the `imageArchResolved[` is just that static string, then outside the double quotes you expand `$lookup`, and then end with the static string `]:+x`, which, again, is not inside `${}`, so it just evaluates as itself. Since `$lookup` is not quoted, if it contains whitespace it will expand to more than one word and bash will complain that `-n` only takes a single argument.

Not for this PR, but it is often safer and clearer to use `[[ ]]` in bash.

Reply:
To make things more concrete:

```sh
$ bash -c 'declare -A imageArchResolved; lookup=debian:latest; if [ -n "imageArchResolved["$lookup"]:+x" ]; then echo "fetch & cache"; else echo "use cache"; fi'
fetch & cache
$ bash -c 'declare -A imageArchResolved; lookup=debian:latest; imageArchResolved+=([$lookup]=386); if [ -n "imageArchResolved["$lookup"]:+x" ]; then echo "fetch & cache"; else echo "use cache"; fi'
fetch & cache
$ bash -c 'declare -A imageArchResolved; lookup=debian:latest; if [ -n "${imageArchResolved[$lookup]:+x}" ]; then echo "fetch & cache"; else echo "use cache"; fi'
use cache
$ bash -c 'declare -A imageArchResolved; lookup=debian:latest; imageArchResolved+=([$lookup]=386); if [ -n "${imageArchResolved[$lookup]:+x}" ]; then echo "fetch & cache"; else echo "use cache"; fi'
fetch & cache
$ bash -c 'declare -A imageArchResolved; lookup=debian:latest; if [ -z "${imageArchResolved[$lookup]:+x}" ]; then echo "fetch & cache"; else echo "use cache"; fi'
fetch & cache
$ bash -c 'declare -A imageArchResolved; lookup=debian:latest; imageArchResolved+=([$lookup]=386); if [ -z "${imageArchResolved[$lookup]:+x}" ]; then echo "fetch & cache"; else echo "use cache"; fi'
use cache
```

Reply:
I had fixed that but pushed to the wrong repo!